pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1085,6 +1088,39 @@ func (h *HTTPArtifact) HasLocation() bool {
 	return h != nil && h.URL != ""
 }
 
+// OSSBucket contains the access information required for interfacing with an OSS bucket
+type OSSBucket struct {
+	// Endpoint is the hostname of the bucket endpoint
+	Endpoint string `json:"endpoint"`
+
+	// Bucket is the name of the bucket
+	Bucket string `json:"bucket"`
+
+	// AccessKeySecret is the secret selector to the bucket's access key
+	AccessKeySecret apiv1.SecretKeySelector `json:"accessKeySecret"`
+
+	// SecretKeySecret is the secret selector to the bucket's secret key
+	SecretKeySecret apiv1.SecretKeySelector `json:"secretKeySecret"`
+
+}
+
+// OSSArtifact is the location of an OSS artifact
+type OSSArtifact struct {
+	OSSBucket `json:",inline"`
+
+	// Key is the key in the bucket where the artifact resides
+	Key string `json:"key"`
+}
+
+func (o *OSSArtifact) String() string {
+	protocol := "https"
+	return fmt.Sprintf("%s://%s/%s/%s", protocol, o.Endpoint, o.Bucket, o.Key)
+}
+
+func (o *OSSArtifact) HasLocation() bool {
+	return o != nil && o.Bucket != ""
```

Comment:
do you need to check Endpoint and/or Key?

Reply:
fixed